keyboards/massdrop/ctrl/keymaps/xanimos/keymap.c

```diff
@@ -0,0 +1,410 @@
+    },
+};
+
+// Runs just one time when the keyboard initializes.
+void matrix_init_user(void) {
+    // Enable or disable debugging
+    debug_enable = false;
+    debug_matrix = false;
+    debug_keyboard = false;
+    debug_mouse = false;
+
+    idle_second_counter = 0;                            // Counter for number of seconds keyboard has been idle.
+    key_event_counter = 0;                              // Counter to determine if keys are being held, neutral at 0.
+    rgb_time_out_seconds = RGB_DEFAULT_TIME_OUT;        // RGB timeout initialized to its default configure in keymap.h
+    rgb_time_out_enable = false;                        // Disable RGB timeout by default. Enable using toggle key.
+    rgb_time_out_user_value = false;                    // Has to have the same initial value as rgb_time_out_enable.
+    rgb_enabled_flag = true;                            // Initially, keyboard RGB is enabled. Change to false config.h initializes RGB disabled.
+    rgb_time_out_fast_mode_enabled = false;             // RGB timeout fast mode disabled initially.
+    rgb_time_out_saved_flag = rgb_matrix_get_flags();   // Save RGB matrix state for when keyboard comes back from ide.
+};
+
+void keyboard_post_init_user(void) {
+    rgb_matrix_enable();
+}
+
+// Runs constantly in the background, in a loop.
+void matrix_scan_user(void) {
+    if(rgb_time_out_enable && rgb_enabled_flag) {
+        // If the key event counter is not zero then some key was pressed down but not released, thus reset the timeout counter.
+        if (key_event_counter) {
+            idle_second_counter = 0;
+        } else if (timer_elapsed(idle_timer) > MILLISECONDS_IN_SECOND) {
+            idle_second_counter++;
+            idle_timer = timer_read();
+        }
+
+        if (idle_second_counter >= rgb_time_out_seconds) {
+            rgb_time_out_saved_flag = rgb_matrix_get_flags();
+            rgb_matrix_set_flags(LED_FLAG_NONE);
+            rgb_matrix_disable_noeeprom();
+            rgb_enabled_flag = false;
+            idle_second_counter = 0;
+        }
+    }
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    static uint32_t key_timer;
+
+    // Increment key event counter for every press and decrement for every release.
+    if (record->event.pressed) {
+        key_event_counter++;
+
+    } else {
+        key_event_counter--;
+    }
+
+    if (rgb_time_out_enable) {
+        idle_timer = timer_read();
+        // Reset the seconds counter. Without this, something like press> leave x seconds> press, would be x seconds on the effective counter not 0 as it should.
+        idle_second_counter = 0;
+        if (!rgb_enabled_flag) {
+            rgb_matrix_enable_noeeprom();
+            rgb_matrix_set_flags(rgb_time_out_saved_flag);
+            rgb_enabled_flag = true;
+        }
+    }
+
+    switch (keycode) {
+        case MD_BOOT:
```

Comment:
You could just use `RESET` here and override the default implemention
```suggestion
        case RESET:
```